sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -543,7 +561,7 @@ case class FileSourceScanExec(
         dataSchema = relation.dataSchema,
         partitionSchema = relation.partitionSchema,
         requiredSchema = requiredSchema,
-        filters = pushedDownFilters,
+        filters = dynamicallyPushedDownFilters,
```

Comment:
One implicit change is, now we will trigger subquery execution when getting the rdd of the file scan node. This worries me a little bit. Can we trigger subquery execution in the AQE framework? e.g. before executing a query stage, transform the scan node and update its `dataFilters` to replace evaluated scalar subqueries with literals.

Reply:
We do not change the timing of trigger for subquery execution. If we call `execute` at `SparkPlan`, then it would always first prepare and call `updateResult` to executing subquery. So it seems there is no change with this pr. This pr only uses the result of subquery and push it down.

Reply:
Could we split the `dynamicallyPushedDownFilters` to `pushedDownFilters` and `pushedDownRuntimeFilters` and combine them here? Something like this: https://github.com/apache/spark/pull/36128/files#diff-089285f1484c1598cb2839b86b6a9e65b98ab5b30462aedc210fe4bbf44cae78R374-R455
This is because `pushedDownFilters` are also used in [`metadata`](https://github.com/apache/spark/blob/abd841bdcabc64d04e0743b0c16443cf0bb66558/sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala#L416), while `pushedDownRuntimeFilters` can only be used here.

Reply:
I did not change pushedDownFilters, it should not be a issue with `metadata`. But it's a pity that we can not make metadata aware of scalar subquery filter.

Reply:
It is not the issue. Just wanted to make it clearer. `dynamicallyPushedDownFilters` also includes `pushedDownFilters`.

Reply:
If we first filter out scalar subqueries and then add them back, then the ordering of pushed filter is changed. e.g.,
`c1 > (select min(x) from t) and c2 > 1` -> `c2 > 1 and c1 > (select min(x) from t)`
I'm not sure if it affects the performance in parquet side. So I simply re-translate the whole data filter.